jira/client.py

```diff
@@ -573,7 +573,22 @@ def __init__(
 
         if server:
             options["server"] = server
+
+        options["async_class"] = None
         if async_:
+            if self._options["async"]:
+                try:
+                    from requests_futures.sessions import FuturesSession
+
+                    options["async_class"] = FuturesSession
+                except ImportError:
+                    msg = (
+                        "async option requires requests-futures to be installed. "
+                        "falling back to synchronous implementation."
+                    )
+                    warnings.warn(msg)
+                    pass
```

Comment:
My preference would be to:

1. do the import check outside the scope of this class, then either set `FutureSession` as `None` if it is not imported or have a boolean variable, eg. `_has_async_support`. 
2. This warning can be emitted in init as you have done if the async option is passed incorrectly. 
Furthermore I would augment this warning message with a suggestion of how to install this extra requirement, as we do have this as an extra, it may be appropriate to say, `this can be installed using the 'async' extra, e.g. pip install jira[async]`
3. I would be tempted to modify the option itself to be False and then no further code changes need to be made in the dependency functions.
4. Code that needs the async may check for the boolean or that the imported class is not None.

Reply:
ive updated the code based on your suggestion. Im unclear on what 3 means.